### PR TITLE
ACAS-633: GCS file support

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
 - Changes to support external file storage for GCS

## Related Issue
ACAS-633

## How Has This Been Tested?
ACAS client tests and UI testing

### CODE REVIEW

File: R/fileRead.R [hunk 1]
```diff
@@ -542,52 +542,6 @@ validateNumeric <- function(inputValue, errorEnv = NULL) {
   }
   return(coercedValue)
 }
-#' File Saving (deprecated)
```
Comment (brianbolt) [inline R/fileRead.R:545]: Wasn't bing used so I removed it.

File: R/batchAnnotation.R [hunk 2]
```diff
@@ -109,7 +117,7 @@ addFileLink <- function(batchCodeList, recordedBy, experiment, lsTransaction,
   }
   locationState <- existingStates[[1]]
   
-  if (racas::applicationSettings$server.service.external.file.type == "blueimp") {
+  if (racas::applicationSettings$server.service.external.file.type %in% c("blueimp", "gcs")) {
```
Comment (brianbolt) [inline R/batchAnnotation.R:120]: This applies to both our known file storage types

File: R/batchAnnotation.R [hunk 3]
```diff
@@ -23,9 +23,11 @@ addFileLink <- function(batchCodeList, recordedBy, experiment, lsTransaction,
     fileName <- basename(reportFilePath)
     
     experimentCodeName <- experiment$codeName
+    entityType <- "experiment"
+    folderPath <- entityFileStorePaths[entityType]
```
Comment (brianbolt) [inline R/batchAnnotation.R:27]: entityFileStorePaths is basically an enum lookup for folder paths in the racas package.

**Review by brianbolt — COMMENTED** [review]
After reviewing my own code here I realize I should probably just call the new ACAS API for moving files around instead of having any knowledge of it in R but at the time of writing this I was trying to make as minimal changes as possible.

File: R/batchAnnotation.R [hunk 4]
```diff
@@ -36,8 +38,14 @@ addFileLink <- function(batchCodeList, recordedBy, experiment, lsTransaction,
...
-      serverFileLocation <- file.path("experiments", experimentCodeName, fileName)
-    } else {
+      serverFileLocation <- file.path(folderPath, experimentCodeName, fileName)
+    } else if (racas::applicationSettings$server.service.external.file.type %in% c("custom", "gcs")) {
+      if(!exists('customSourceFileMove') || is.null(customSourceFileMove)) {
+        stop(paste0("customSourceFileMove has not been defined in customFunctions.R"))
+      }
+      serverFileLocation <- customSourceFileMove(reportFilePath, recordedBy, fileName = fileName, entityType = entityType, entity = experiment, 
+                                                deleteOldFile = TRUE, additionalPath = NA)
+    } else {ß
```
Comment (bffrost) [inline R/batchAnnotation.R:48]: What is that funky `ß` character on this line?

File: R/batchAnnotation.R (hunk 2 above)
Comment (bffrost) [reply R/batchAnnotation.R:120]: Not to "custom" though?

**Review by bffrost — APPROVED** [review]
LGTM, nice refactor to remove magic strings, and add in "gcs" type wherever "custom" is.

**Pull request closed without merging**